### PR TITLE
Distinguish message for external macros depending on error level

### LINKED ISSUE #57716: Distinguish between error vs. warning for external macros
I often see compiler notes for *warnings* accompanied with this message:
```
   = note: this error originates in a macro outside this <...>
```

would it be more sensible and correct to say `this _warning_ originates...`?

Very superficial but I find seeing "error" in compilation logs catches my eye far too often and
its only a warning. Happy to submit PR if it is a valid change.

### PR DESCRIPTION
**Pull request by JohnTitor**
fixes #57716

(I picked you because assigned to this issue.)
r? @estebank

### CODE REVIEW

File: src/librustc_errors/emitter.rs [hunk 1]
```diff
@@ -856,18 +858,28 @@ impl EmitterWriter {
     fn fix_multispans_in_std_macros(&mut self,
                                     span: &mut MultiSpan,
                                     children: &mut Vec<SubDiagnostic>,
+                                    level: &Level,
                                     backtrace: bool) {
         let mut spans_updated = self.fix_multispan_in_std_macros(span, backtrace);
         for child in children.iter_mut() {
             spans_updated |= self.fix_multispan_in_std_macros(&mut child.span, backtrace);
         }
+        // here
```
Comment (Centril) [inline src/librustc_errors/emitter.rs]: What is the purpose of this comment?

Comment (JohnTitor) [reply src/librustc_errors/emitter.rs]: Oops, I forgot removing the mark for myself, I'll remove :)

**Comment by Centril** [comment]
r? @Centril 

Can you also squash the commits into one?
With that, r=me once travis is green.

**Comment by JohnTitor** [comment]
@Centril Okay, I squashed the commits, thank you.

**Pull request merged**